src/Server/TCPHandler.cpp

```diff
@@ -1026,7 +1026,17 @@ bool TCPHandler::receivePacket()
             return false;
 
         case Protocol::Client::Cancel:
+        {
+            /// For testing connection collector.
+            const Settings & settings = query_context->getSettingsRef();
+            if (settings.sleep_in_receive_cancel_ms.totalMilliseconds())
+            {
+                std::chrono::milliseconds ms(settings.sleep_in_receive_cancel_ms.totalMilliseconds());
+                std::this_thread::sleep_for(ms);
```

Comment:
Not important, but maybe it worth using sleepForMilliseconds(), to handle EINTR

```suggestion
                sleepForMilliseconds(settings.sleep_in_receive_cancel_ms.totalMilliseconds());
```

(here and below)